jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -512,33 +512,27 @@ private void checkAccess(Class<?> targetClass, String name, int memberModifiers,
 		 * @throws IllegalAccessException If the {@link Class} is not accessible from {@link #accessClass}. 
 		 */
 		private void checkClassAccess(Class<?> targetClass) throws IllegalAccessException {
-			/*[IF Sidecar19-SE]*/
-			checkClassModuleVisibility(accessMode, accessClass.getModule(), targetClass);
-			/*[ENDIF]*/
-			if (NO_ACCESS != accessMode) {
-				/* target class should always be accessible to the lookup class when they are the same class */
-				if (accessClass == targetClass) {
-					return;
-				}
-				
-				int modifiers = targetClass.getModifiers();
-				
-				/* A protected class (must be a member class) is compiled to a public class as
+			int targetModifiers = targetClass.getModifiers();
+			boolean accessible = false;
+			if (Modifier.isPublic(targetModifiers) || Modifier.isProtected(targetModifiers)) {
```

Comment:
Classes at the VM level are either public or package.  Are there tests that should `protected` should be treated like public?

Reply:
There is a JTReg test for Lookup on protected classes: 
https://github.com/eclipse/openj9/issues/3301#issuecomment-435983845.  I suggest we concentrate on closing that, since it fixes a JTReg test, then revisit this.